pkg/cmd/adm/register_member.go

```diff
@@ -120,59 +119,224 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+	generatedKubeConfigFormatted, err := clientcmd.Write(*generatedKubeConfig)
+	if err != nil {
+		return err
 	}
-	term.Printlnf("Command to be called: bash %s\n", strings.Join(args, " "))
-	bash := newCommand("bash", args...)
-	bash.Stdout = os.Stdout
-	bash.Stderr = os.Stderr
-	return bash.Run()
-}
 
-func downloadScript(term ioutils.Terminal) (*os.File, error) {
-	resp, err := http.Get(AddClusterScriptURL)
+	// Create or Update the secret on the clusterJoinTo
+	secretName := secretName(SANamespacedName, joiningClusterDetails.OperatorNamespace, joiningClusterDetails.Name)
+	term.Printlnf("creating %s secret with name %s/%s", joiningClusterType, clusterJoinToDetails.OperatorNamespace, secretName)
+	kubeConfigSecret := &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: clusterJoinToDetails.OperatorNamespace}}
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToDetails.KubeClient, kubeConfigSecret, func() error {
+
+		// update the secret label
+		labels := kubeConfigSecret.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[toolchainv1alpha1.ToolchainClusterLabel] = joiningClusterDetails.Name
+		kubeConfigSecret.Labels = labels
+
+		// update the kubeconfig data
+		kubeConfigSecret.StringData = map[string]string{
+			"kubeconfig": string(generatedKubeConfigFormatted),
+		}
+
+		return nil
+	})
+
 	if err != nil {
-		return nil, errs.Wrapf(err, "unable to get add-script.sh")
-	}
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return nil, fmt.Errorf("unable to get add-script.sh - response status %s", resp.Status)
+		return err
 	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil {
-			term.Printlnf(err.Error())
+	term.Println("Secret successfully reconciled")
+	term.Printlnf("operation", op)
+
+	// TODO -- temporary logic
+	// The creation of the toolchaincluster is just temporary until we implement https://issues.redhat.com/browse/KUBESAW-44,
+	// the creation logic will be moved to the toolchaincluster_resource controller in toolchain-common and will be based on the secret created above.
+	//
+	// create/update toolchaincluster on the clusterJoinTo
+	term.Printlnf("creating ToolchainCluster representation of %s in %s:", joiningClusterType, clusterJoinToDetails.Name)
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: joiningClusterDetails.Name, Namespace: clusterJoinToDetails.OperatorNamespace}}
+	op, err = controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToDetails.KubeClient, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels["namespace"] = joiningClusterDetails.OperatorNamespace
+		if joiningClusterType == "member" {
+			labels["cluster-role.toolchain.dev.openshift.com/tenant"] = ""
+		}
```

Comment:
I have a feeling that this is an legacy field, that we don't rely on this label anymore. @metlos migrated this into SPC CR.

Reply:
correct . I see that the capacity manager relies on the placementRoles from the SPC. I've dropped the label in https://github.com/kubesaw/ksctl/pull/53/commits/d80e98f0291693cbcd784d095d703b0f287ade38 

Thanks !